src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/KestrelHttpParser.cs

```diff
@@ -390,142 +388,127 @@ public KestrelHttpParser(IKestrelTrace log)
             }
         }
 
-        private unsafe int IndexOf(byte* pBuffer, int index, int length, byte value)
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private unsafe int IndexOfNameEnd(byte* pBuffer, int index, int length)
```

Comment:
Did this and some other stuff in https://github.com/aspnet/KestrelHttpServer/pull/1455